### PR TITLE
[UX] Reset autostop timer whenever autostop config changes on remote machine

### LINKED ISSUE #1511: [Autostop] Need to reset the timer for every execution stage
Our user observed that the spot controller is autostopped, during the submission of a spot job (during setting up the controller). It is possible because we only reset the autostop timer, when the controller process is submitted to the ray job.

One way to mitigate it is to reset the autostop timer at the beginning of every execution stage. cc @concretevitamin

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #1511

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [x] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
  - [x] `pytest tests/test_smoke.py::test_autostop`
  - [x] `pytest tests/test_smoke.py::test_autodown`
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/cli.py [hunk 1]
```diff
@@ -2559,17 +2559,12 @@ def autostop(
 
     - The cluster has restarted.
 
-    - An autostop is set when there is no active setting. (Namely, either
-      there's never any autostop setting set, or the previous autostop setting
-      was canceled.) This is useful for restarting the autostop timer.
+    - A new autostop idle time is set.
```
Comment (concretevitamin) [inline sky/cli.py:2562]: "new": What if there was an existing N minutes, and users submit an autostop command with N again?

File: sky/cli.py [hunk 2]
```diff
@@ -2559,17 +2559,12 @@ def autostop(
...
 
-    Example: say a cluster without any autostop set has been idle for 1 hour,
```
Comment (concretevitamin) [inline sky/cli.py:2566]: Keep/update this scenario?

File: sky/cli.py [hunk 3]
```diff
@@ -2559,17 +2559,12 @@ def autostop(
...
-    then an autostop of 30 minutes is set. The cluster will not be immediately
-    autostopped. Instead, the idleness timer only starts counting after the
-    autostop setting was set.
-
-    When multiple autostop settings are specified for the same cluster, the
-    last setting takes precedence.
+    Example: say a cluster with autostop set to be 2 hours has been idle for 1
+    hour, then autostop is reset to 30 minutes. The cluster will not be
+    immediately autostopped. Instead, the idleness timer only restart counting
+    after autostop setting was reset.
```
Comment (concretevitamin) [inline sky/cli.py:2567]:
```suggestion
    Example: say a cluster with autostop set to 2 hours has been idle for 1
    hour, then autostop is reset to 30 minutes. The cluster will not be
    immediately autostopped. Instead, the idleness timer restarts counting
    when the second autostop setting of 30 minutes was submitted.
```

File: sky/skylet/autostop_lib.py [hunk 4]
```diff
@@ -54,13 +54,11 @@ def get_autostop_config() -> AutostopConfig:
...
-    prev_autostop_config = get_autostop_config()
     configs.set_config(_AUTOSTOP_CONFIG_KEY, pickle.dumps(autostop_config))
     logger.debug(f'set_autostop(): idle_minutes {idle_minutes}, down {down}.')
-    if (prev_autostop_config.autostop_idle_minutes < 0 or
-            prev_autostop_config.boot_time != psutil.boot_time()):
-        # Either autostop never set, or has been canceled. Reset timer.
-        set_last_active_time_to_now()
+    # Reset timer whenever the autostop is set on the machine, i.e. the idle
+    # time will be counted from now.
+    set_last_active_time_to_now()
```
Comment (concretevitamin) [inline sky/skylet/autostop_lib.py:61]:
```suggestion
    # Reset timer whenever an autostop setting is submitted, i.e. the idle
    # time will be counted from now.
    set_last_active_time_to_now()
```

File: sky/skylet/autostop_lib.py [hunk 5]
```diff
@@ -54,13 +54,11 @@ def get_autostop_config() -> AutostopConfig:
 def set_autostop(idle_minutes: int, backend: Optional[str], down: bool) -> None:
     boot_time = psutil.boot_time()
     autostop_config = AutostopConfig(idle_minutes, boot_time, backend, down)
-    prev_autostop_config = get_autostop_config()
     configs.set_config(_AUTOSTOP_CONFIG_KEY, pickle.dumps(autostop_config))
     logger.debug(f'set_autostop(): idle_minutes {idle_minutes}, down {down}.')
-    if (prev_autostop_config.autostop_idle_minutes < 0 or
-            prev_autostop_config.boot_time != psutil.boot_time()):
```
Comment (concretevitamin) [inline sky/skylet/autostop_lib.py:61]: Do we handle the previous logic `prev_autostop_config.boot_time != psutil.boot_time()`? Does this handle the case when a cluster is restarted, the idleness timer should reset?

**Review by concretevitamin — COMMENTED** [review]
Thanks, some comments.

File: sky/cli.py (hunk 1 above)
Comment (Michaelvll) [reply sky/cli.py]: Removed "new" to be more clear. Thanks!

File: sky/skylet/autostop_lib.py (hunk 5 above)
Comment (Michaelvll) [reply sky/skylet/autostop_lib.py:61]: Yes, in the current code, we will reset the active time in any cases when the `set_autostop` function is called, including the restarted case.

File: sky/cli.py (hunk 2 above)
Comment (Michaelvll) [reply sky/cli.py:2566]: The scenario shoud be similar as the new scenario we have. Do you think it is needed to keep this?

Comment (concretevitamin) [reply sky/cli.py:2566]: I think it'd help the users/readers. Even though the technical impl is the same, it's "setting autostop for the first time" vs. "resetting autostop" scenarios.

**Review by concretevitamin — APPROVED** [review]
LGTM

**Pull request merged**